build cannot find gopass and go-flags packages

`README.md` states:

> After doing `go get github.com/shazow/ssh-chat` on this repo, you should be able to run a command like:
>
> `$ ssh-chat --verbose --bind ":22" --identity ~/.ssh/id_dsa`

But doing that, at least with with an empty `$GOPATH` workspace, fails to build `ssh-chat`:

    $ brew install go
    ...
    $ go version
    go version go1.4.2 darwin/amd64
    $ rm -rf $GOPATH
    $ mkdir $GOPATH
    $ go get github.com/shazow/ssh-chat
    $ ls $GOPATH
    pkg/ src/

Note the lack of a `bin/` directory. Investigating, I find these build errors:

    $ cd $GOPATH/src/github.com/shazow/ssh-chat
    $ make
    go build -ldflags "-X main.buildCommit `git describe --long --tags --dirty --always`" ./cmd/ssh-chat
    cmd/ssh-chat/key.go:10:2: cannot find package "code.google.com/p/gopass" in any of:
    	/usr/local/Cellar/go/1.4.2/libexec/src/code.google.com/p/gopass (from $GOROOT)
    	$GOPATH/src/code.google.com/p/gopass (from $GOPATH)
    cmd/ssh-chat/cmd.go:15:2: cannot find package "github.com/jessevdk/go-flags" in any of:
    	/usr/local/Cellar/go/1.4.2/libexec/src/github.com/jessevdk/go-flags (from $GOROOT)
    	$GOPATH/src/github.com/jessevdk/go-flags (from $GOPATH)
    make: *** [ssh-chat] Error 1

The packages it cannot locate are needed by the `cmd/ssh-chat` modules.

I figured out that one way to address this is by invoking `go get` recursively:

    $ rm -rf $GOPATH
    $ mkdir $GOPATH
    $ go get github.com/shazow/ssh-chat/...
    $ ls $GOPATH
    bin/ pkg/ src/
    $ ls $GOPATH/bin
    ssh-chat*
    $ $GOPATH/bin/ssh-chat -help
    Usage:
      ssh-chat [OPTIONS]
    ...

I'm new to Go, so this took me a while to figure out, and I don't know if it's the best solution. But it seems like `README.md` should be updated to invoke `go get` recursively, at least.

--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ SRCS = %.go
 
 all: $(BINARY)
 
-$(BINARY): **/**/*.go **/*.go *.go
+$(BINARY): deps **/**/*.go **/*.go *.go
 	go build -ldflags "-X main.buildCommit `git describe --long --tags --dirty --always`" ./cmd/ssh-chat
 
 deps:

--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@ $(BINARY): **/**/*.go **/*.go *.go
 	go build -ldflags "-X main.buildCommit `git describe --long --tags --dirty --always`" ./cmd/ssh-chat
 
 deps:
-	go get .
+	go get ./...
 
 build: $(BINARY)
 

--- a/README.md
+++ b/README.md
@@ -49,7 +49,7 @@ Help Options:
   -h, --help       Show this help message
 ```
 
-After doing `go get github.com/shazow/ssh-chat` on this repo, you should be able
+After doing `go get github.com/shazow/ssh-chat/...` on this repo, you should be able
 to run a command like:
 
 ```

--- a/auth.go
+++ b/auth.go
@@ -54,7 +54,7 @@ func NewAuth() *Auth {
 }
 
 // AllowAnonymous determines if anonymous users are permitted.
-func (a Auth) AllowAnonymous() bool {
+func (a *Auth) AllowAnonymous() bool {
 	return a.whitelist.Len() == 0
 }
 

--- a/chat/command.go
+++ b/chat/command.go
@@ -62,7 +62,7 @@ func (c Commands) Alias(command string, alias string) error {
 
 // Run executes a command message.
 func (c Commands) Run(room *Room, msg message.CommandMsg) error {
-	if msg.From == nil {
+	if msg.From() == nil {
 		return ErrNoOwner
 	}
 

